OptimoveCore/Sources/Classes/Storage/StorageFacade.swift

```diff
@@ -99,6 +101,29 @@ extension UserDefaults: KeyValueStorage {
     }
 }
 
+public final class InMemoryStorage: KeyValueStorage {
+    private var storage = [StorageKey: Any]()
```

Comment:
Are there any thread-safety concerns w/ this in-memory store?

(not familiar with how it's being used, or even if stale reads could matter)

Reply:
Adding an access queue for being thread-safety.

Reply:
this in memory storage was for delivered events, right, currently not used?

Reply:
> this in memory storage was for delivered events, right, currently not used?

it is in use for the optimove part, keeping `[.tenantToken, .version]` in memory, where `version` is a config name from the optimove config.